### DISCUSSION TITLE
Why did I need to swap `$model` and `$relation` in model relationship controller?

### DISCUSSION THREAD

**Discussion by ChaDonSom** [tailflow/laravel-orion#221] [Help] [answered]
Hey guys, I ran into something recently that maybe someone could help me figure out.

I have `Job` and `Task`, where Job hasMany Tasks.

I added a controller like this:

```php
<?php

namespace App\Http\Controllers\Api;

use App\Job;
use App\Task;
use Illuminate\Http\Request;
use Illuminate\Database\Eloquent\Builder as EloquentBuilder;
use Orion\Http\Controllers\Controller;

class JobTaskController extends Controller {
    protected $model = Job::class;

    protected $relation = 'tasks';

    /**
     * Builds Eloquent query for fetching entity(-ies).
     *
     * @param Request $request
     * @param array $requestedRelations
     * @return Builder
     */
    protected function buildFetchQuery(Request $request, array $requestedRelations): EloquentBuilder {
        $query = parent::buildFetchQuery($request, $requestedRelations);

        $query->whereHas('job', fn($query) => $query->where('jobs.company_id', '=', $request->user()->company_id));

        return $query;
    }

    public function exposedScopes(): array
    {
        return ['withCalculatedQuantity', 'withTotalHours', 'withTotalOvertimeHours'];
    }
}
```

And registered it in `api.php` like this:

```php
    Orion::resource('jobs',      JobController::class);
    Orion::resource('tasks',     TaskController::class);
    Orion::hasManyResource('jobs', 'tasks', JobTaskController::class);
```

But I found the API responses are giving me Jobs rather than Tasks? (at this endpoint: `https://96d3-216-237-193-16.ngrok-free.app/api/jobs/14170/tasks/search`)

But when I switch the `$model` and `$relation` properties, like this:

```php
    protected $model = Task::class;

    protected $relation = 'job';
```

... then it works?

But it doesn't seem like this is exactly the right answer either, so I'm a little bit confused. Is there anything obvious I'm messing up here?

Thank you for checking this out!

**Comment by rishpandey** [comment, marked as answer]
For anyone looking for the answer to this. One is supposed to extend Orion\Http\Controllers\RelationController (not Controller) for controllers associated with relations.
